src/uct/tcp/tcp_sockcm_ep.h

```diff
@@ -0,0 +1,57 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2019.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#include "tcp_listener.h"
+
+
+typedef enum uct_tcp_sockcm_ep_state {
+    UCT_TCP_SOCKCM_EP_ON_SERVER      = UCS_BIT(0),
+    UCT_TCP_SOCKCM_EP_ON_CLIENT      = UCS_BIT(1),
+    UCT_TCP_SOCKCM_EP_INIT           = UCS_BIT(2),
+    UCT_TCP_SOCKCM_EP_CONNECTED      = UCS_BIT(3), /* connect()/accept()
+                                                      completed successfully */
+    UCT_TCP_SOCKCM_EP_SENDING        = UCS_BIT(4),
+    UCT_TCP_SOCKCM_EP_DATA_SENT      = UCS_BIT(5),
+    UCT_TCP_SOCKCM_EP_DATA_RECEIVING = UCS_BIT(6),
+    UCT_TCP_SOCKCM_EP_DATA_RECVED    = UCS_BIT(7),
+    UCT_TCP_SOCKCM_EP_DISCONNECTING  = UCS_BIT(8),  /* uct_ep_disconnect was
+                                                       called on the ep.
+                                                       this ep is not
+                                                       necessarily disconnected
+                                                       yet */
+    UCT_TCP_SOCKCM_EP_FAILED         = UCS_BIT(9)
+} uct_tcp_sockcm_ep_state_t;
+
+
+/**
+ * TCP sockcm endpoint communication context
+ */
+typedef struct uct_tcp_sockcm_ep_ctx {
```

Comment:
this structure looks redundant, can we move it directly to EP? maybe use anonymous structure to unite the fields.
BTW I can't find how is it used (or going to be used) for send/recv or send only? comment to structure contradicts to fields comments

Reply:
it isn't used yet, but i plan to use it for send and recv operations. i put it here in the header anyway for review. 
why do you think it's redundant?

Reply:
> why do you think it's redundant?

this is a part of EP and not used (maybe for now) out of the EP scope.

> i plan to use it for send and recv operations.

then pls fix comments "send" -> "send/recv"

Reply:
will fix the comment. i think that since these are 3 fields are relevant to a certain functionality of the ep, they can be united to one struct.

Reply:
yes, but there is no need to define new type. it can be anonymous.
```
typedef struct ep {
    struct {
    ...
    } comm_ctx;
} ep_t;
```